pkg/apply/prune/prune.go

```diff
@@ -163,7 +163,10 @@ func (p *Pruner) Prune(
 }
 
 // removeInventoryAnnotation removes the `config.k8s.io/owning-inventory` annotation from pruneObj.
-func (p *Pruner) removeInventoryAnnotation(obj *unstructured.Unstructured) error {
+func (p *Pruner) removeInventoryAnnotation(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
Do we need to return the cloned resource here? Unlike in the apply case, the object we mutate here isn't actually an input to the library but something that we read from the cluster.

Reply:
The obj comes from the Prune objs input, which comes from PruneTask.Objects. I don't think we want to modify PruneTask.Objects while the task is running.

Reply:
Another option would be to create a PATCH directly that removes the annotation, rather than modifying the object and then making a PATCH, but that would be a bigger change to the code here.

Reply:
Agree that it is probably better to avoid mutating the resources. PATCH could be a good option, but agree that should be a separate PR.